crates/ide-assists/src/handlers/remove_parentheses.rs

```diff
@@ -0,0 +1,91 @@
+use syntax::{ast, AstNode};
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: remove_parentheses
+//
+// Removes redundant parentheses.
+//
+// ```
+// fn main() {
+//     _ = $0(2) + 2;
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     _ = 2 + 2;
+// }
+// ```
+pub(crate) fn remove_parentheses(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<()> {
+    let parens = ctx.find_node_at_offset::<ast::ParenExpr>()?;
+
+    let cursor_in_range =
+        parens.l_paren_token()?.text_range().contains_range(ctx.selection_trimmed())
+            || parens.r_paren_token()?.text_range().contains_range(ctx.selection_trimmed());
+    if !cursor_in_range {
+        return None;
+    }
+
+    let expr = parens.expr()?;
+
+    let parent = ast::Expr::cast(parens.syntax().parent()?);
+    let is_ok_to_remove = expr.precedence() >= parent.as_ref().and_then(ast::Expr::precedence);
```

Comment:
As a side-note: this check is _completely_ broken. The question of "do we need parentheses here?" is actually quite complicated (with Rust at least), and I've fallen into a rabbit hole here, trying to fix this.

I have _some_ progress, but the code is terrible and still doesn't handle everything, hopefully I'll be able to fix that sooner rather than later :(